source/controllers/tests/testImpedanceController.cpp

```diff
@@ -0,0 +1,51 @@
+#include "controllers/impedance/Impedance.hpp"
+#include "state_representation/Robot/JointState.hpp"
+#include "state_representation/Robot/JointTorques.hpp"
+#include "state_representation/Space/Cartesian/CartesianState.hpp"
+#include "state_representation/Space/Cartesian/CartesianWrench.hpp"
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Dense>
+#include <gtest/gtest.h>
+
+using namespace controllers::impedance;
+using namespace StateRepresentation;
+
+TEST(TestCartesianImpedance, PositiveNos) {
+  Impedance<CartesianState> impedance_controller(Eigen::MatrixXd::Identity(6, 6),
+                                                 Eigen::MatrixXd::Identity(6, 6),
+                                                 Eigen::MatrixXd::Identity(6, 6));
+  // set up a desired state6d
+  CartesianState desired_state("test");
+  desired_state.set_linear_velocity(Eigen::Vector3d(1, 0, 0));
+  // set up a real state
+  CartesianState feedback_state("test");
+  feedback_state.set_orientation(Eigen::Quaterniond(0, 0, 1, 1));
```

Comment:
sort of unrelated question - is the orientation quaternion regularly normalised in any of the state operations? Because at the moment this is of course not a unit quaternion. I would assume it is normalised at the appropriate step, but it's worth checking those assumptions!

Reply:
Quaternions are indeed normalized at each `set_orientation` call. This is why I am not bothering putting a unit quaternion.